### PR TITLE
Akka IO statsd client

### PR DESCRIPTION
**Pull request by janm399**
Much improved statsd client. Naive performance measurements put it at 8x the speed of the `java.net`-based one. Do not merge just yet; will complete over the weekend.

### CODE REVIEW

**Comment by janm399** [comment]
~~@archena, can you include the DD extensions for tags? Also, can we have a sample statsd server for tests?~~

All done in the latest push.

**Comment by archena** [comment]
@janm399 perhaps compare performance for 'connected' udp: https://github.com/archena/statsd-akka/blob/master/src/main/scala/statsdclient/statsd.scala

Now that tags are added, I suppose this doesn't work with plain statsd? when I tried sending DD tags to a statsd server it didn't like it at all.

**Comment by janm399** [comment]
@archena see the latest

File: output-statsd/src/main/scala/org/eigengo/monitor/output/statsd/AkkaIOStatsdCounterInterface.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  import StatsdActor._
+
+  def recordExecutionTime(aspect: String, duration: Int, tags: String*): Unit =
+    statsd ! ExecutionTime(aspect, duration, tags)
+
+  def recordGaugeValue(aspect: String, value: Int, tags: String*): Unit =
+    statsd ! Gauge(aspect, value, tags)
+
+  def decrementCounter(aspect: String, tags: String*): Unit =
```
Comment (archena) [inline output-statsd/src/main/scala/org/eigengo/monitor/output/statsd/AkkaIOStatsdCounterInterface.scala:42]: Also decrement counter with delta?

Comment (janm399) [inline output-statsd/src/main/scala/org/eigengo/monitor/output/statsd/AkkaIOStatsdCounterInterface.scala:42]: We haven't needed it yet in the aspects, so I'll hang on until it is required. We'd just be writing code for the sake of tests.

File: output-statsd/src/test/scala/org/eigengo/monitor/output/statsd/PerformanceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+      val aioTime = timed(count)(aio.incrementCounter("aio"))
+      // wait for all messages
+      Thread.sleep(2000)
+
+      // we should see 2 entries, each with $count values
+      import scala.collection.JavaConversions._
+      map.size() mustEqual 2
+      // we allow half of the UDP datagrams to be lost
+      map.elements().toList.forall(_.intValue() > count / 2)
```
Comment (hughsimpson) [inline output-statsd/src/test/scala/org/eigengo/monitor/output/statsd/PerformanceSpec.scala:61]: Is it worth logging what proportion of datagrams are actually lost here, so we can see that by looking at the TravisCI monitor output? Or is that just more meaningless numbers?

**Pull request merged**